command-line-local-shortener/main.go

```diff
@@ -75,12 +76,12 @@ func postReq(c *gin.Context) {
 		c.String(http.StatusBadRequest, "enter a url to shorten")
 		return
 	} else {
-		result := dataProccess(longURL)
-		if result != "" {
-			c.String(http.StatusOK, "Your new link is: http://localhost:8080/"+result)
-
-		} else {
+		result, err := dataProccess(longURL)
+		if err != nil {
 			c.String(http.StatusBadRequest, "invalid URL")
+		} else {
+			c.String(http.StatusOK, fmt.Sprintf(
```

Comment:
You're going to want to replace this with a `c.JSON` eventually. This way the frontend will receive a Javascript object back instead of just plain text.